Task3CountingAllSymbolsInFile.c

```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+int main()
+{
+    printf("This program shows frequencies of all symbols found in file input.txt!\n");
+    FILE* file = fopen("input.txt", "r");
+    if (file == NULL)
+    {
+        printf("Error opening file input.txt :(");
+        return -1;
+    }
+    unsigned char arrayOfChars[256] = {};
+    int arrayOfCounters[256] = {0};
+
+    for (int i = 0; i < 256; i++)
+    {
+        arrayOfChars[i] = (unsigned char)i;
+    }
+    while(!feof(file))
+    {
+        unsigned char symbol = (unsigned char)fgetc(file);
+
+        int codeOfSymbol = (int)symbol;
+        arrayOfCounters[codeOfSymbol] += 1;
+
+    }
+    arrayOfCounters[255] = 0;
```

Comment:
```suggestion
    while(!feof(file))
    {
        int codeOfSymbol = fgetc(file);
        if (codeOfSymbol != -1)
        {
            ++arrayOfCounters[codeOfSymbol];
        }
    }
```
Здесь забавно получается. `fgetc` берет следующий символ из файла и переносит указатель на этот символ.
`feof` сообщит о том, что файл исчерпан, только когда указатель будет правее последнего символа в файле.
Вот и получается, что на последней итерации указатель в файле указывает на последний символ, `feof` запускает нас в тело цикла, `fgetc` возвращает следующий символ (символа нет, поэтому `-1`), при привидении типов `-1` превращается в `255`, с которым вы боретесь после цикла.